docs/architecture/domain.md

```diff
@@ -0,0 +1,116 @@
+## Definition Model
+
+![Definition](assets/definition-model.png)
+
+### Pipeline
+
+A pipeline represents a collection of [jobs] that share the same source code. These jobs are executed in the order defined by the [workflow]. Only one job is required to be defined in every pipeline:
+
+ - `main` - Job that runs on each change to the source code.  _Note: this is also the same job that runs on proposed changes via GitHub Pull Requests_
+
+### Workflow
+
+Workflow is the defined order that [jobs] will execute in. Jobs can be executed in parallel, series, and a combination of the two to allow for all possibilities. All jobs executed in a given workflow run share:
+
+ - Source code checked out to the same git commit from the `main` [build] that triggered this
+ - Access to the [metadata] from the `main` [build] that triggered this
```

Comment:
Does that mean that they can modify the `metadata` from the `main` build? Or is just that they have RO access to this metadata?

If the second, maybe rephrase it as:
"A copy of metadata from the main build"

Also can a human trigger this job? If so, what metadata/git commit does it use? Is it the last successful `main` build?